lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -151,6 +154,7 @@ public ParquetReader(
         this.fileCreatedBy = requireNonNull(fileCreatedBy, "fileCreatedBy is null");
         requireNonNull(columnFields, "columnFields is null");
         this.columnFields = ImmutableList.copyOf(columnFields);
+        this.appendRowNumberColumn = appendRowNumberColumn;
```

Comment:
We also have code in `io.trino.plugin.hive.parquet.ParquetPageSource#getRowIndexColumn` for appending row number column. Can we avoid duplicating this functionality ?